### ISSUE TITLE
Apache Lexer raises NoMethodError error

### ISSUE THREAD

**Issue by benbalter** [jneen/rouge#385]
### The error:

```
NoMethodError: undefined method `<<' for nil:NilClass
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/lexer.rb:392:in `block in lex'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:436:in `yield_token'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:333:in `token'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/lexers/apache.rb:43:in `block (2 levels) in <class:Apache>'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:305:in `instance_exec'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:305:in `block in step'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:287:in `each'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:287:in `step'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:268:in `stream_tokens'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/lexer.rb:388:in `lex'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/formatters/html.rb:50:in `each'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/formatters/html.rb:50:in `stream_untableized'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/formatters/html.rb:43:in `stream'
	from /usr/local/opt/rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/formatter.rb:33:in `format'
```

### Steps to reproduce (per the readme)

1. Create a file "test.txt" with contents "test"
2. `require 'rouge'`
3. `source = File.read("test.txt")`
4. `formatter = Rouge::Formatters::HTML.new(css_class: 'highlight')`
5. `lexer = Rouge::Lexers::Apache.new`
6. `formatter.format(lexer.lex(source))`

**Comment by parkr** [comment]
If you turn on debugging for the lexer, you see:

```irb
>> require 'rouge'
>> formatter = Rouge::Formatters::HTML.new(css_class: 'highlight')
>> lexer = Rouge::Lexers::Apache.new(debug: true)
>> formatter.format(lexer.lex("text\n"))
lexer: apache
stack: ["root"]
stream: "text\n"
  entering mixin whitespace
  trying #<Rule /\#.*?\n/>
  trying #<Rule /[\s\n]+/m>
  exiting  mixin whitespace
  trying #<Rule /(<\/?)(\w+)/>
  trying #<Rule /\w+/>
    got "text"
NoMethodError: undefined method `qualname' for nil:NilClass
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:435:in `yield_token'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:333:in `token'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/lexers/apache.rb:43:in `block (2 levels) in <class:Apache>'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:305:in `instance_exec'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:305:in `block in step'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:287:in `each'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:287:in `step'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/regex_lexer.rb:268:in `stream_tokens'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/lexer.rb:388:in `lex'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/formatters/html.rb:50:in `each'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/formatters/html.rb:50:in `stream_untableized'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/formatters/html.rb:43:in `stream'
	from /Users/parkr/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rouge-1.10.1/lib/rouge/formatter.rb:33:in `format'
	from (irb):7
	from /Users/parkr/.rbenv/versions/2.3.0/bin/irb:11:in `<main>'
```

**Comment by julp** [comment]
This issue was already reported by #246, #350 and #300 (I think they are the same even if results are different)

It should be fixed by master (so in next release) by #301

**Comment by parkr** [comment]
@julp You rock! Thank you for finding those other issues. Glad to hear it's all sorted on master.
